jdbc_utils.py

```diff
@@ -25,18 +25,28 @@ def to_jdbc_url(dbi_url):
 
 def remove_credentials(error):
     """
-    For PostgreSQL errors, strip out everything except error code and description.
+    For PostgreSQL login errors, strip out everything except error code and description.
     This prevents usernames from being printed.
     For other Flyway-specific errors, go ahead leave them intact.
     """
     code_lookup = {
-        '08001': 'Connection attempt failed',
-        '28P01': 'Password authentication failed',
-        '42501': 'Insufficient priveledge'
+        '08000': 'connection_exception',
+        '08001': 'sqlclient_unable_to_establish_sqlconnection',
+        '08003': 'connection_does_not_exist',
+        '08004': 'sqlserver_rejected_establishment_of_sqlconnection',
+        '08006': 'connection_failure',
+        '08007': 'transaction_resolution_unknown',
+        '08P01': 'protocol_violation',
+        '28000': 'invalid_authorization_specificiation',
+        '28P01': 'invalid_password',
+        '42000': 'syntax_error_or_access_rule_violation',
+        '42501': 'insufficient_privilege',
+        '42601': 'syntax_error'
     }
     match = re.search(r'.*(SQL State  : )+(?P<error>[a-zA-Z0-9]{0,5})', error)
     if match:
         error_code = match.group('error')
```

Comment:
Since we’re using the fallback error for our .get, can we streamline our “if” logic here?

Reply:
@lbeaufort I don't know a priori all the error messages flyway throws, so if there is no match, i still want to throw the error. There are also logically equivalent ways to handle that case as well, but let me know what you are thinking. Thank you!

Reply:
Below is a short discussion via comments. 
```python
match = re.search(r'.*(SQL State  : )+(?P<error>[a-zA-Z0-9]{0,5})', error)
"""
regex match successful indicating we know how to extract error code (do ALL errors satisfy the above regular expression?
"""
if match: 
    error_code = match.group('error')
    return 'PostgreSQL error code {}: {}'.format(
        error_code, code_lookup.get(error_code, error)) #get the error code, but if we didn't account for it in advance, throw the whole error, but still requires the match
"""
if the error is something not expected or in a format not known in advance, then return that error
"""
else:
    return error
```

Reply:
@jason-upchurch looks good, thanks for taking a look!